diagnostic_common_diagnostics/src/diagnostic_common_diagnostics/cpu_monitor.py

```diff
@@ -34,35 +34,45 @@
 
 # \author Rein Appeldoorn
 
-import rospy
-from diagnostic_updater import DiagnosticTask, Updater
-from diagnostic_msgs.msg import DiagnosticStatus
+import collections
+import socket
 
 import psutil
-import socket
+import rospy
+from diagnostic_msgs.msg import DiagnosticStatus
+from diagnostic_updater import DiagnosticTask, Updater
 
 
 class CpuTask(DiagnosticTask):
-    def __init__(self, warning_percentage):
+    def __init__(self, warning_percentage, window):
         DiagnosticTask.__init__(self, "CPU Information")
         self._warning_percentage = int(warning_percentage)
+        self._readings = collections.deque(maxlen=window)
+
+    def _get_average_reading(self):
+        def avg(lst):
+            return float(sum(lst)) / len(lst)
```

Comment:
Could you please also handle the len(lst) = 0 case just to be on the safe side?

Reply:
Done